### PR TITLE
HPCC-24523 Add filecompcopy and httpcall_xmlpost to spray class

### PR DESCRIPTION
**Pull request by AttilaVamos**
Add //class=spray tags to:
- filecompcopy.ecl
- httpcall_jsonpost.ecl
- httpcall_xmlpost.ecl

test cases to exclude them from Regression Suite on a containerized
environment.

add missing copyright information to
- httpcall_multiheader.ecl

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@jakesmith @gfortil please review.

File: testing/regress/ecl/httpcall_multiheader.ecl [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+//class=spray
+//class=roxieserviceaccess
```
Comment (jakesmith) [inline testing/regress/ecl/httpcall_multiheader.ecl:19]: why has 'roxieserviceaccess' been added, does this query access roxie?

**Comment by jakesmith** [comment]
@AttilaVamos - 1 question, and related, I think PR comment is misleading:
```
Add //class=spray and //class=roxieserviceaccess tags to:

filecompcopy.ecl
httpcall_jsonpost.ecl
httpcall_multiheader.ecl
httpcall_xmlpost.ecl
```

filecompcopy isn't/doesn't have 'roxieserviceaccess',
and not sure it should have been added to httpcall_multiheader.

**Comment by richardkchapman** [comment]
@AttilaVamos This seems to be stalled?

Should it be targeting master?

File: testing/regress/ecl/httpcall_multiheader.ecl (hunk 1 above)
Comment (AttilaVamos) [reply testing/regress/ecl/httpcall_multiheader.ecl:19]: It was a mistake. Removed (as well as the "//class=spray" based on this isn't perform any spray operation)

**Comment by AttilaVamos** [comment]
@jakesmith updated, please re-review.

**Comment by jakesmith** [comment]
@AttilaVamos looks fine, but the PR commit needs changing (doesn't reflect the changes) - and the commit message not quite right (says httpcall_xmlpost.ecl changed), but hasn't.

Please change commit and PR comments and squash.

**Comment by AttilaVamos** [comment]
@jakesmith updated and squashed,

**Comment by jakesmith** [comment]
@AttilaVamos - looks good.

**Comment by jakesmith** [comment]
@richardkchapman - ready to merge.

**Pull request merged**